src/st/no3/e_fire_warg.c

```diff
@@ -24,73 +24,82 @@ static void func_801CC5A4(Entity* entity, u8 count, u8 params, s32 xDist,
     }
 }
 
-static void func_801CC6F8(Entity* entity) {
+static void func_801CC6F8(Entity* self) {
     u16 distance = GetDistanceToPlayerX();
-    bool var_s1;
+    bool buttNearScreenEdge;
 
-    entity->ext.et_801CC820.unk86 = 0x100;
-    entity->velocityX = 0;
+    self->velocityX = 0;
+    self->ext.fireWarg.unk86 = 0x100;
 
-    if (entity->params != 0) {
+    if (self->params) {
         SetStep(12);
         return;
     }
 
-    var_s1 = 0;
-
-    if (entity->facingLeft != 0) {
-        var_s1 = entity->posX.i.hi >= 0x71;
-    } else if (entity->posX.i.hi < 0x90) {
-        var_s1 = 1;
+    // This will test if we're facing left, with high X, or facing right with
+    // low X.
+    buttNearScreenEdge = false;
+    if (self->facingLeft) {
+        if (self->posX.i.hi > 0x70) {
+            buttNearScreenEdge = true;
+        }
+    } else {
+        if (self->posX.i.hi < 0x90) {
+            buttNearScreenEdge = true;
+        }
     }
 
     if (distance < 0x70) {
-        if (!(Random() & 3) && !var_s1) {
+        if (!(Random() & 3) && !buttNearScreenEdge) {
             SetStep(9);
-            return;
+        } else {
+            SetStep(6);
+            PlaySfxPositional(0x783);
+            self->ext.fireWarg.unk80 = 0x20;
         }
-    } else if ((Random() & 3) && !var_s1) {
-        SetStep(9);
-        if (!(Random() & 3)) {
-            entity->ext.et_801CC820.unk86 = 0;
+    } else {
+        if ((Random() & 3) && !buttNearScreenEdge) {
+            SetStep(9);
+            if (!(Random() & 3)) {
+                self->ext.fireWarg.unk86 = 0;
+            }
+        } else {
+            SetStep(6);
+            PlaySfxPositional(0x783);
+            self->ext.fireWarg.unk80 = 0x20;
         }
-        return;
     }
-
-    SetStep(6);
-    PlaySfxPositional(0x783);
-    entity->ext.et_801CC820.unk80 = 0x20;
 }
 
 // Only called by the EntityFireWarg
-static void func_801CC820(Entity* entity) {
+static void func_801CC820(Entity* self) {
     u16 distance;
 
-    if (entity->facingLeft == GetSideToPlayer()) {
-        if (entity->params == 0) {
-            SetStep(5);
-        } else {
+    if ((self->facingLeft == GetSideToPlayer()) & 1) {
```

Comment:
Great question! That comes from our old friend the PSP :) https://decomp.me/scratch/F8pqQ

Nice job spotting that one.

Reply:
Ahh yes.  I should have known.

Reply:
Hmmm, it occurs to me that the " & 1" might have been a condition that the compiler determined would always be true. It could have been something like `    if ((self->facingLeft == GetSideToPlayer()) & (7 == 7)) {` or something silly like that. The PSP compiler resolved the condition to always be 1, while the PS1 compiler fully removed the second conditional. Oh well, we'll probably never know.

Reply:
I wonder if it was some leftover debug thing where they wanted to force skip the branch on occasion

ie

#define DO_SKIP 1

if(..snip.. & (DO_SKIP ==0)) 

or some nonsense like that.  I know that I occasionally was doing that when I was learning to do some emulation stuff to skip to where I needed to get to in a rom execution.